src/include/sof/audio/asrc/IasSrcFarrow.h

```diff
@@ -0,0 +1,681 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2012 Intel Corporation. All rights reserved.
+
+/*
+ * @file     IasSrcFarrow.h
+ * @author   Intel Transportation Solutions Division (TSD)
+ * @date     27. Nov 2017
+ * @version  1.3.5
+ * @brief    API header containing sample rate converter struct and interface
+ *           functions.
+ *
+ * @mainpage Hifi3 Implementation of the Intel TSD Sample Rate Converter
+ *
+ * The sample rate converter is based on the so-called Farrow structure.
+ * It supports multi-channel operation, and the PCM samples can be
+ * represented by int16_t or int32_t values.
+ *
+ * The sample rate converter can be applied for transmit and receive
+ * use cases.  To support both directions, the sample rate converter
+ * offers two modes of operation: push-mode and pull-mode.
+ * <p>
+ * @image html PushMode-PullMode-UseCase_150dpi.png
+ * "Push-mode vs. pull-mode operation"
+ * @image rtf  PushMode-PullMode-UseCase_150dpi.png
```

Comment:
this should be converted to a REST documentation.

Reply:
I need to get familiar with code originated documentation. Also, I don't have this picture. It wasn't part of the OS PDT process.

Reply:
Not done. I think I will need to redraw the picture myself when SOF Docs documentation is done for this component.